lib/screens/player_screen.dart

```diff
@@ -199,6 +241,97 @@ class _PlayerScreenContent extends StatelessWidget {
       ),
     );
   }
+
+  PageRouteBuilder _buildSlideRouteTransition(Widget sourceWidget, Widget targetWidget, {
+    RouteSettings? routeSettings,
+  }) {
+    return PageRouteBuilder(
+      settings: routeSettings,
+      pageBuilder: (context, animation, secondaryAnimation) => targetWidget,
+      transitionsBuilder: (context, animation, secondaryAnimation, child) {
+        const curve = Curves.ease;
+        const beginEnter = Offset(1.0, 0.0);
+        const endEnter = Offset.zero;
+        const beginExit = Offset(0.0, 0.0);
+        const endExit = Offset(-1.0, 0.0);
+
+        final tweenEnter = Tween(begin: beginEnter, end: endEnter);
+        final tweenExit = Tween(begin: beginExit, end: endExit);
+        final curvedAnimation = CurvedAnimation(
+          parent: animation,
+          curve: curve,
+        );
+
+        return Stack(
+          children: [
+            SlideTransition(
+              position: tweenExit.animate(curvedAnimation),
+              child: sourceWidget,
+            ),
+            SlideTransition(
+              position: tweenEnter.animate(curvedAnimation),
+              child: child,
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _buildBottomActions(BuildContext context, PlayerHideableController controller) {
+    return Consumer(
+      builder: (context, ref, child) {
+
+        final metadata = ref.watch(currentTrackMetadataProvider).value;
+
+        final isLyricsLoading = metadata?.item.mediaStreams?.any((e) => e.type == "Lyric") == null;
```

Comment:
I'm not sure these flag names make sense?  Also, why are manually checking the streams when we have the hasLyrics attribute?

Reply:
Not sure what happened there. Should be better now